torch/_inductor/utils.py

```diff
@@ -108,11 +120,11 @@ def sympy_dot(seq1, seq2):
     return sympy.expand(sum(a * b for a, b in zip(seq1, seq2)))
 
 
-def unique(it):
+def unique(it: Iterable[T]) -> Iterable[T]:
     return {id(x): x for x in it}.values()
```

Comment:
I know this isn't part of the PR, but I am really curious why it wasn't implemented a set. Probably a hack to maintain ordering.

Reply:
probably to keep it deterministic, yes